mdstat.go

```diff
@@ -0,0 +1,161 @@
+package procfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	statuslineRE = regexp.MustCompile(`(\d+) blocks .*\[(\d+)/(\d+)\] \[[U_]+\]`)
+	buildlineRE  = regexp.MustCompile(`\((\d+)/\d+\)`)
+)
+
+// MDStatus holds info parsed from /proc/mdstat.
+type MDStatus struct {
+	// Name of the device.
+	MDName string
+	// activity-state of the device.
+	ActivityState string
+	// Number of active disks.
+	DisksActive int64
+	// Total number of disks the device consists of.
+	DisksTotal int64
+	// Number of blocks the device holds.
+	BlocksTotal int64
+	// Number of blocks on the device that are in sync.
+	BlocksSynced int64
+}
+
+// ParseMdstat parses an mdstat-file and returns a struct with the relevant infos.
+func (fs FS) ParseMdstat() (mdstates []MDStatus, err error) {
```

Comment:
The reason some - not all - types have a Proc prefix is that there is a difference between `/proc/stat` and `/proc/$pid/stat`. Some types only apply to processes and need to be distinguised. The name of the package is also `procfs` and not `proc` so I don't follow your argumentation of an anti-pattern here. Interested to hear other naming suggestions though.

As `mdstat` is not bound to specific processes and generally available under `/prod/mdstat` it should not have a `Proc` Prefix.

I agree that `MDStat` seems to match the procfs naming better than `MDStatus`.